src/queue.c

```diff
@@ -5458,6 +5491,9 @@ _dispatch_worker_thread(void *context)
 	dispatch_root_queue_context_t qc = dq->do_ctxt;
 	dispatch_pthread_root_queue_context_t pqc = qc->dgq_ctxt;
 
+	int pending = (int)os_atomic_dec2o(qc, dgq_pending, relaxed);
```

Comment:
I know it's pre-existing but dgq_pending should be made an int to avoid this tasteless cast.